File: csharp/Behavioral/14-Command/Command/PauseJobCommand.cs
public class PauseJobCommand : ICommand
{
    private readonly JobCard _jobCard;
    private string           _previousStatus = string.Empty;

    public PauseJobCommand(JobCard jobCard) => _jobCard = jobCard;

    public void Execute()
    {
        _previousStatus = _jobCard.Status;
        _jobCard.Status = "Paused";
        Console.WriteLine($"  Paused {_jobCard.JobId}. Status → Paused");
    }

    public void Undo()
    {
        _jobCard.Status = _previousStatus;
        Console.WriteLine($"  Undo pause: {_jobCard.JobId} Status → {_previousStatus}");
    }
}

File: typescript/Behavioral/14-Command/Command/PauseJobCommand.ts
import { ICommand } from "./ICommand";
import { JobCard } from "../Model/JobCard";

export class PauseJobCommand implements ICommand {
  private previousStatus: string = "";

  constructor(private readonly jobCard: JobCard) {}

  execute(): void {
    this.previousStatus = this.jobCard.status;
    this.jobCard.status = "Paused";
    console.log(`  Paused ${this.jobCard.jobId}. Status → Paused`);
  }

  undo(): void {
    this.jobCard.status = this.previousStatus;
    console.log(
      `  Undo pause: ${this.jobCard.jobId} Status → ${this.previousStatus}`,
    );
  }
}
